src/Voron/Impl/LowLevelTransaction.cs

```diff
@@ -140,33 +131,28 @@ public Size AdditionalMemoryUsageSize
         internal ref TransactionHeader TransactionHeader => ref _txHeader;
 
         private readonly HashSet<PageFromScratchBuffer> _transactionPages;
-        private readonly HashSet<long> _freedPages;
-        private readonly List<PageFromScratchBuffer> _unusedScratchPages;
-
+        private bool _hasFreePages;
 
-        private readonly StorageEnvironmentState _state;
+        private const int RevertedScratchPageMarker = -0xDEAD;
```

Comment:
We use it _only_ in `Debug.Assert()`. Shouldn't we use it as assignment somewhere?

Reply:
Unused